control/control_loop.go

```diff
@@ -113,31 +114,40 @@ func createLoop(logger logging.Logger, cfg Config, m Controllable) (*Loop, error
 			l.activeBackgroundWorkers.Add(1)
 			utils.ManagedGo(func() {
 				b := b
-				nInputs := len(b.ins)
 				close(waitCh)
 				for {
-					sw := make([]*Signal, nInputs)
-					for i, c := range b.ins {
+					sw := []*Signal{}
+					s := []*Signal{}
+					for _, c := range b.ins {
 						r, ok := <-c
 						if !ok {
 							b.mu.Lock()
 							for _, out := range b.outs {
 								close(out)
 							}
-							// logger.Debugf("Closing outs for block %s %+v\r\n", b.blk.Config(ctx).Name, r)
 							b.outs = nil
 							b.mu.Unlock()
 							return
 						}
-						if len(r) == 1 {
-							sw[i] = r[0]
+						for j := 0; j < len(r); j++ {
+							if r[j] != nil {
+								sw = append(sw, r[j])
+							}
+						}
+						// TODO(npmenard) do we want to support multidimentional signals?
+						//nolint: makezero
```

Comment:
I'm pretty sure this line doesn't go here: the next line down has nothing to do with pre-allocating array capacity or pre-allocating array values.

edit: I see where this was copied and pasted from. I suspect the way we could have removed it in the past was to change the initialization of `sw` from `make([]*Signal, nInputs)` to `make([]*Signal, 0, nInputs)`. but the way you changed it should work, too. We can remove this nolint line entirely.